app/models/Book.js

```diff
@@ -0,0 +1,14 @@
+class Book {
+  constructor(id, createdAt, updatedAt, title, author, publisher, yearOfPublication, numberOfPages) {
+    this.id = id;
+    this.createdAt = createdAt;
+    this.updatedAt = updatedAt;
+    this.title = title;
+    this.author = author;
+    this.publisher = publisher;
+    this.yearOfPublication = yearOfPublication;
+    this.numberOfPages = numberOfPages;
+  }
+
+  // Add validation methods and relationships here if necessary
+}
```

Comment:
Consider adding validation methods to ensure data integrity, such as checking for non-empty strings for title, author, and publisher, and valid numbers for yearOfPublication and numberOfPages.